File: Examples/PHP/Containers/Laravel-MongoDB-REST/app/Services/MongoDB/CRUD/IMongoDB_CRUD_Client.php
<?php

namespace App\Services\MongoDB\CRUD;

interface IMongoDB_CRUD_Client
{
    public function ping(): bool;
    public function createAsync(string $dbName, string $collectionName, \MongoDB\Model\BSONDocument $document): string;
    public function readAsync(string $dbName, string $collectionName, string $documentId): \MongoDB\Model\BSONDocument;
    public function updateAsync(string $dbName, string $collectionName, string $documentId, \MongoDB\Model\BSONDocument $updateFields): ?\MongoDB\Model\BSONDocument;
    public function deleteAsync(string $dbName, string $collectionName, string $documentId): bool;
}

File: Examples/CSharp/Containers/MongoDB-REST/MongoDB.REST/MongoDB/CRUD/IMongoDB_CRUD_Client.cs
using MongoDB.Driver;
using MongoDB.Bson;
using System.Threading;
using System.Threading.Tasks;

namespace MongoDB.CRUD.Client;

public interface IMongoDB_CRUD_Client : IDisposable
{
    Task Ping();
    Task<ObjectId?> CreateAsync(string dbName, string collectionName, BsonDocument document, CancellationToken cancellationToken = default);
    Task<BsonDocument?> ReadAsync(string dbName, string collectionName, ObjectId documentId, CancellationToken cancellationToken = default);
    Task<BsonDocument?> UpdateAsync(string dbName, string collectionName, ObjectId documentId, BsonDocument updateFields, CancellationToken cancellationToken = default);
    Task<bool> DeleteAsync(string dbName, string collectionName, ObjectId documentId, CancellationToken cancellationToken = default);
}